ome_zarr/utils.py

```diff
@@ -46,6 +46,26 @@ def info(path: str, stats: bool = False) -> Iterator[Node]:
         yield node
 
 
+def validate(path: str, warnings: bool) -> Iterator[Node]:
+    """
+    Validate OME-NGFF data
+
+    All :class:`Nodes <ome_utils.reader.Node>` that are found from the given path will
+    be visited recursively.
+    """
+    zarr = parse_url(path)
+    assert zarr, f"not a zarr: {zarr}"
+    reader = Reader(zarr)
+    for node in reader():
+        if not node.specs:
+            print(f"not an ome-zarr node: {node}")
+            continue
+
+        if hasattr(node, "validate"):
+            node.validate(warnings)
+        yield node
```

Comment:
So, if you just do this: does it `report` by default (and validate)?
```
$ ome_zarr info image.zarr
```
Then you can disable both with: ?
```
$ ome_zarr info image.zarr --no-validate --no-report
```

Reply:
Just my 2cents here: I think the cleanest design would be to implement `def visit(path: str, func: callable)` that evaluates `func` for each node. This could then be used to implement both `info` and `validate` without much code duplication, i.e. both `info` and `validate` would just use `visit` with an appropriate function.

This has the added advantage that it allows further customization by users. If you go for this functionality I would also adopt the "usual" syntax for visit in python: If `func` returns something other than `None`, `visit` terminates and returns it.